js-core/homeworks/homework-12/src/main.js

```diff
@@ -91,36 +90,34 @@ function Human(object) {
     this.gender = object.gender;
     this.height = object.height;
     this.weigth = object.weigth;
-    this.inheritHuman = function() {
-        return this;
-    }
 }
 
-Human.prototype.Worker = function(object) {
+function Worker(object) {
     this.company = object.company;
     this.salary = object.salary;
     this.toWork = function() {
         return 'work';
     }
-    this.inheritProperties = this.inheritHuman();
+    Human.call(this, object)
+    Worker.prototype = new Human(object);
```

Comment:
it's a bad idea to rewrite CLASS prototype every-time when new instance is creating